### PR TITLE
explain squashing

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -49,6 +49,35 @@ When you send a PR, just make sure that:
...
+Replace all `pick` prefixes by `fixup` (or `f`) **except the first commit** of the list.
+
+Save and quit the editor.
+
+After that, all your commits where squashed into the first one and the commit message of the first commit.
+
+If you would like to rename your commit message type:
+
+```bash
+git commit --amend
```
Comment (Soullivaneuh) [inline CONTRIBUTING.md]: Well, this method works, but you can do all in the same type.

On rebase, replace `pick` by `reword` on the commit you want to rewrite and an another editor will appear to edit the comment.

But I don't know. This method looks better for newbies.

**Comment by rande** [comment]
I will prefere a link to an official documentation either from github or from a blog post

**Comment by Soullivaneuh** [comment]
> I will prefere a link to an official documentation either from github or from a blog post

Well, the issue is I didn't find any official documentation with an easy to understand logic..

**Comment by OskarStark** [comment]
> Well, the issue is I didn't find any official documentation with an easy to understand logic..

yes...

The Symfony guys removed the squashing thing from their docs, because the do it on their own with a merge tool

**Comment by Soullivaneuh** [comment]
> with a merge tool

Yeah I heard about that... But can't find where. Which tools are they using by curiosity?

**Comment by OskarStark** [comment]
> Which tools are they using by curiosity?

don't know, maybe some self developed piece of code

**Pull request merged**